tools/timelock_proposal/mcm_with_timelock.go

```diff
@@ -0,0 +1,149 @@
+package timelock_proposal
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/errors"
+	owner "github.com/smartcontractkit/ccip-owner-contracts/tools/gethwrappers"
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/mcms_proposal"
+)
+
+var ZERO_HASH = common.BytesToHash([]byte{})
+
+type MCMSWithTimelockChainMetadata struct {
+	mcms_proposal.ChainMetadata
+	TimelockAddress common.Address `json:"timelockAddress"`
+}
+
+type TimelockOperation string
+
+const (
+	Schedule TimelockOperation = "schedule"
+	Cancel   TimelockOperation = "cancel"
+	Bypass   TimelockOperation = "bypass"
+)
+
+type MCMSWithTimelockProposal struct {
+	mcms_proposal.Proposal
+
+	Operation TimelockOperation `json:"operation"` // Always 'schedule', 'cancel', or 'bypass'
+
+	// TODO: this should be configurable as a human-readable string
+	// i.e. 1d, 1w, 1m, 1y
+	MinDelay string `json:"minDelay"`
+
+	// Overridden: Map of chain identifier to chain metadata
+	ChainMetadata map[string]MCMSWithTimelockChainMetadata `json:"chainMetadata"`
+
+	// Overridden: Operations to be executed after wrapping in a timelock
+	Transactions []BatchChainOperation `json:"transactions"`
+}
+
+func (m *MCMSWithTimelockProposal) Validate() error {
+	if err := m.Proposal.Validate(); err != nil {
+		return err
+	}
+
+	switch m.Operation {
+	case Schedule, Cancel, Bypass:
```

Comment:
Do we need this validation? The proposal should fail to Unmarshal because it's a defined type `TimelockOperation`

Reply:
will it definitely fail to unmarshal? either way, will add a test for it and if thats the case then I'll remove this validation

Reply:
Hm, following up on this, atleast in the unit tests, I can pass in an arbitrary strings and it doesn't seem to error unless I call validate. Maybe json.Unmarshal will catch it if that happens but some of these will be programatically constructed so probably wanna catch this error sooner than when reading from file